policies/recipes/recipedb.go

```diff
@@ -15,25 +15,69 @@
 package recipes
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"os"
 	"strconv"
 	"strings"
 )
 
+var (
+	dbPath = "~/placeholder_location/swr.json"
+)
+
 // RecipeDB represents local state of installed recipes.
-type RecipeDB struct{}
+type RecipeDB struct {
+	recipes map[string]Recipe
+}
 
-func newRecipeDB() RecipeDB {
-	return RecipeDB{}
+func newRecipeDB() (RecipeDB, error) {
+	db := RecipeDB{}
+	f, err := os.Open(dbPath)
+	if err != nil {
+		return RecipeDB{}, err
+	}
+	defer f.Close()
+	bytes, err := ioutil.ReadAll(f)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return RecipeDB{recipes: make(map[string]Recipe)}, nil
+		}
+		return RecipeDB{}, err
+	}
+	if err := json.Unmarshal(bytes, &db); err != nil {
+		return RecipeDB{}, err
+	}
+	return db, nil
 }
 
 // GetRecipe returns the Recipe object for the given recipe name.
 func (db *RecipeDB) GetRecipe(name string) (Recipe, bool) {
-	return Recipe{}, false
+	r, ok := db.recipes[name]
+	return r, ok
```

Comment:
I went for the strategy where it will always write state to disk but only read from disk once. What's the concurrency model for this, should it read from disk all the time and does it need any locking mechanisms?

Reply:
it should not be possible for the same recipe name to be present in multiple policies, so reading once is probably ok.

will consider the write locking scenario, but it at least doesn't need it -yet-.